src/perfrecord.cpp

```diff
@@ -190,3 +224,40 @@ void PerfRecord::sendInput(const QByteArray& input)
     Q_ASSERT(m_perfRecordProcess);
     m_perfRecordProcess->write(input);
 }
+
+bool PerfRecord::checkFilePermissions(const QString &filePath)
+{
+    QFileInfo outputFile(filePath);
+    QString username = currentUsername();
+
+    if (outputFile.owner() != username) {
+        QString sudoExe = sudoUtil();
+        if (sudoExe.isEmpty() || username.isEmpty()) {
+            return false;
+        }
+
+        QString chownCommand = sudoExe + QStringLiteral(" -u root chown ") +
```

Comment:
I'd also prefer using a QStringList and separate literals for the individual commands arguments like above